internal/set.go

```diff
@@ -30,3 +36,176 @@ func (s IntSet) Remove(e int) {
 func (s IntSet) Count() int {
 	return len(s)
 }
+
+// Same determines whether two sets are backed by the same store. In the
+// current implementation using hash maps it makes use of the fact that
+// hash maps (at least in the gc implementation) are passed as a pointer
+// to a runtime Hmap struct.
+//
+// A map is not seen by the runtime as a pointer though, so we cannot
+// directly compare the sets converted to unsafe.Pointer and need to take
+// the sets' addressed and dereference them as pointers to some comparable
+// type.
+func Same(s1, s2 Set) bool {
+	return *(*uintptr)(unsafe.Pointer(&s1)) == *(*uintptr)(unsafe.Pointer(&s2))
```

Comment:
Okay, I checked out the golang-nuts thread. I definitely agree with the motivation. It shouldn't need to take len(map) to determine if the maps are equal. We clearly need an O(1) solution. But I don't see what's wrong with reflect

http://play.golang.org/p/6O2Lb6gWU5

Seems much better than going to unsafe.